src/layers/core/polygon-layer/polygon-layer.js

```diff
@@ -73,14 +73,14 @@ export default class PolygonLayer extends CompositeLayer {
     }
   }
 
-  _onHoverSubLayer(info) {
-    info.object = (info.object && info.object.feature) || info.object;
-    this.props.onHover(info);
-  }
+  _onPickSubLayer(handler, info) {
+    Object.assign(info, {
+      layer: this,
+      // override object with picked feature
+      object: (info.object && info.object.feature) || info.object
```

Comment:
2. Where I see the most potential in moving this to the base layer is not having to manually bind the callbacks and override sublayer props in `renderLayers`. Is this something `CompositeLayer` can do automatically without knowing about the sublayers?